data/transactions/logic/parsing.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package logic
+
+import (
+	"encoding/base32"
+	"encoding/base64"
+	"encoding/binary"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/algorand/avm-abi/abi"
+	"github.com/algorand/go-algorand/data/basics"
+)
+
+// AppCallBytes represents an encoding and a value of an app call argument.
+type AppCallBytes struct {
+	Encoding string `codec:"encoding"`
+	Value    string `codec:"value"`
+}
+
+// NewAppCallBytes parses an argument of "form encoding:value" to AppCallBytes.
```

Comment:
form -> from ?

Reply:
Should say: `argument of the form "encoding:value"` The misplaced quotes make it hard to read.  Will fix.